cli/pkg/stack_mgr/stack_service.go

```diff
@@ -342,33 +342,24 @@ func (s *StackService) CollectStackInfo(ctx context.Context, listAll bool, app s
 	stackInfos := make([]*StackInfo, len(stackNames))
 	sort.Strings(stackNames)
 	for i, name := range stackNames {
-		i, name := i, name // https://golang.org/doc/faq#closures_and_goroutines
-		g.Go(func() error {
-			stackInfo, err := stacks[name].GetStackInfo(ctx)
-			if err != nil {
-				log.Warnf("unable to get stack info for %s: %s (likely means the deploy failed the first time)", name, err)
-				if !diagnostics.IsInteractiveContext(ctx) {
-					stackInfos[i] = &StackInfo{
-						Name:    name,
-						Status:  "error",
-						Message: err.Error(),
-					}
+		stackInfo, err := stacks[name].GetStackInfo(ctx)
```

Comment:
Behind the scenes, this func validates the tfe token and prompts the login. If executed in a go routine, interaction with the token refresher is not possible.

Reply:
sounds like the enforceClient is not threadsafe. I made this a concurrent operation because GetStackInfo is kind of a slow operation and when lots of stacks exist, it can be pretty time consuming to query all the info. I think if we make the enforceClient function thread safe, I would prefer to keep this in a goroutine. I think doing so will also open the door to using more concurrency later too. Let me experiment so I can provide an alternate solution for you review

Reply:
enforceClient is not threadsafe, as it interacts with the stdin.

Reply:
The only way to work around it is to enforce the client before the TFE operations, but it's not always a linear process as TFE interactions are hidden behind layers of indirection.

Reply:
https://github.com/chanzuckerberg/happy/pull/1814